File: Examples/Ruby/Cells/post_update_worksheet_cell_style.rb
require 'aspose_cells_cloud'

class Cell

  include AsposeCellsCloud
  include AsposeStorageCloud

  def initialize
    #Get App key and App SID from https://cloud.aspose.com
    AsposeApp.app_key_and_sid("APP_KEY", "APP_SID")
    @cells_api = CellsApi.new  
  end

  def upload_file(file_name)
    @storage_api = StorageApi.new 
    response = @storage_api.put_create(file_name, File.open("../data/" << file_name,"r") { |io| io.read } )
  end

  def post_update_worksheet_cell_style
    file_name = "myWorkbook.xlsx"
    upload_file(file_name)

    sheet_name = "Sheet1" 
    cell_name = "A3"
    cell_style = Style.new
    font = Font.new
    font.double_size = 20
    font.is_bold = true
    font.is_italic = true
    font.is_strikeout = true
    font.size = 15

    cell_style.font = font
    cell_style.shrink_to_fit = true

    background_color = Color.new
    background_color.r = "255"
    background_color.g = "255"
    background_color.b = "255"
    cell_style.background_color = background_color

    response = @cells_api.post_update_worksheet_cell_style(file_name, sheet_name, cell_name, cell_style)
  end

end

cell = Cell.new()
puts cell.post_update_worksheet_cell_style

File: Examples/PHP/Cells/PostUpdateWorksheetCellStyle.php
//ExStart:
<?php
require_once realpath(__DIR__ . '/..') . '/vendor/autoload.php';
require_once realpath(__DIR__ . '/..') . '/Utils.php';

use Aspose\Cells\CellsApi;
use Aspose\Cells\AsposeApp;

class Cell {

	public $cells;

	public function __construct() {
		AsposeApp::$appSID = Utils::appSID;
		AsposeApp::$apiKey = Utils::apiKey;
		$this->cells = new CellsApi();
	}

	public function postUpdateWorksheetCellStyle() {
        // Upload file to Aspose Cloud Storage
        $fileName = "myWorkbook.xlsx";
        Utils::uploadFile($fileName);

        $body = '{
                    "Font": {
                      "DoubleSize": 20.0,
                      "IsBold": true,
                      "IsItalic": true,
                      "IsStrikeout": true,
                      "Size": 15
                    },
                    "BackgroundColor": {
                        "R": "255",
                        "G": "255",
                        "B": "255"
                    },
                    "ShrinkToFit": true
                }';
        $result = $this->cells->PostUpdateWorksheetCellStyle($name=$fileName, $sheetName="Sheet1", $cellName="A3", $storage = null, $folder = null, $body);
        print_r($result);
    }
}

$cell = new Cell();
$cell->postUpdateWorksheetCellStyle();

?>
//ExEnd:
